src/Compilers/Core/Portable/EncodedStringText.cs

```diff
@@ -8,7 +8,7 @@
 
 namespace Microsoft.CodeAnalysis.Text
 {
-    internal static class EncodedStringText
+    public static class EncodedStringText
```

Comment:
I think it would be better (discoverability-wise) to expose a new ```FromEncodedString``` factory on ```StringText```, rather than exposing an entire class.